src/HealthChecks.AzureServiceBus/DependencyInjection/AzureServiceBusHealthCheckBuilderExtensions.cs

```diff
@@ -17,31 +17,36 @@ public static class AzureServiceBusHealthCheckBuilderExtensions
         /// </summary>
         /// <param name="builder">The <see cref="IHealthChecksBuilder"/>.</param>
         /// <param name="connectionString">The azure event hub connection string.</param>
-        /// <param name="eventHubName">The azure event hub name.</param>
+        /// <param name="eventHubName">The azure event hub name.. Required when connection string  is scoped to the whole namespace</param>
         /// <param name="name">The health check name. Optional. If <c>null</c> the type name 'azureeventhub' will be used for the name.</param>
         /// <param name="failureStatus">
         /// The <see cref="HealthStatus"/> that should be reported when the health check fails. Optional. If <c>null</c> then
         /// the default status of <see cref="HealthStatus.Unhealthy"/> will be reported.
         /// </param>
         /// <param name="tags">A list of tags that can be used to filter sets of health checks. Optional.</param>
         /// <param name="timeout">An optional System.TimeSpan representing the timeout of the check.</param>
-        /// <param name="requiresSession">An optional boolean flag that indicates whether session is enabled on the queue or not. Defaults to false.</param>
         /// <returns>The <see cref="IHealthChecksBuilder"/>.</returns>
-        public static IHealthChecksBuilder AddAzureEventHub(this IHealthChecksBuilder builder, string connectionString, string eventHubName, string name = default, HealthStatus? failureStatus = default, IEnumerable<string> tags = default, TimeSpan? timeout = default)
+        public static IHealthChecksBuilder AddAzureEventHub(this IHealthChecksBuilder builder, string connectionString, string eventHubName = null, string name = default, HealthStatus? failureStatus = default, IEnumerable<string> tags = default, TimeSpan? timeout = default)
         {
             return builder.Add(new HealthCheckRegistration(
                 name ?? AZUREEVENTHUB_NAME,
-                sp => new AzureEventHubHealthCheck(connectionString, eventHubName),
+                sp =>
+                {
+                    return string.IsNullOrWhiteSpace(eventHubName)
+                        ? new AzureEventHubHealthCheck(connectionString)
+                        : new AzureEventHubHealthCheck(connectionString, eventHubName);
```

Comment:
For me, specify will be mandatory,.. but probably I'm missing something!

Reply:
What do you consider mandatory? `eventHubName`? That's actually the point of the pull request because that notition of information can already be part of the connection string, allowing you to give access to just that one event hub rather than all of them in the namespace

Reply:
See https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/pull/472/files#diff-aeba8fbe8f3a9590317be1212801c074R8